sdk/src/main/java/me/uport/sdk/Uport.kt

```diff
@@ -27,33 +24,16 @@ object Uport {
 
     private lateinit var prefs: SharedPreferences
 
-    private lateinit var accountCreator: AccountCreator
-
-    private var defaultAccountHandle = ""
-
-    var defaultAccount: Account?
-        get() = accountStorage?.get(defaultAccountHandle)
-        set(value) {
-            val newDefault = value?.copy(isDefault = true)
-            @Suppress("LiftReturnOrAssignment")
-            if (newDefault == null) {
-                accountStorage?.delete(defaultAccountHandle)
-                defaultAccountHandle = ""
-            } else {
-                val oldAccounts = accountStorage
-                        ?.all()
-                        ?.map { it.copy(isDefault = false) }
-                        ?: emptyList()
-                accountStorage?.upsertAll(oldAccounts + newDefault)
-                defaultAccountHandle = newDefault.handle
-            }
-        }
+    private lateinit var accountCreator: HDAccountCreator
 
-    private var accountStorage: AccountStorage? = null
+    private var accountStorage: SharedPrefsAccountStorage? = null
 
-    private const val UPORT_CONFIG: String = "uport_sdk_prefs"
+    @Suppress("UnsafeCast")
+    var defaultAccount: HDAccount?
+        get() = accountStorage?.getDefaultAccount() as HDAccount?
+        set(value) { }
 
-    private const val OLD_DEFAULT_ACCOUNT: String = "default_account"
+    private const val UPORT_CONFIG: String = "uport_sdk_prefs"
```

Comment:
It's probably going to be easier to migrate stuff if the preferences don't overlap.
`uport_sdk_prefs` should remain the legacy configuration and a new one that uses AccountHolder should be used once migration is done.